fix(curriculum): updates `body-parser` readme with better explanation for `extended` config (#42740)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/05-apis-and-microservices/basic-node-and-express/use-body-parser-to-parse-post-requests.md

# --instructions--

> Install the `body-parser` module in your `package.json`. Then, `require` it at the top of the file. Store it in a variable named `bodyParser`. The middleware to handle urlencoded data is returned by `bodyParser.urlencoded({extended: true})`. Pass to `app.use()` the function returned by the previous method call. As usual, the middleware must be mounted before all the routes which need it.

Reviewer: ```suggestion
Install the `body-parser` module in your `package.json`. Then, `require` it at the top of the file. Store it in a variable named `bodyParser`. The middleware to handle urlencoded data is returned by `bodyParser.urlencoded({extended: true})`. Pass the function returned by the previous method call to `app.use()`. As usual, the middleware must be mounted before all the routes that depend on it.
```

Author: Hi @alirezaghey I have commited the suggestion you have asked for.

**Note:** `extended` is a configuration option that tells `body-parser` which parsing needs to be used. When `extended=false` it uses the classic encoding `querystring` library. When `extended=true` it uses `qs` library for parsing. 

> When using `extended=false`, values can be only strings or arrays. The object returned when using `querystring` does not prototypically inherit from the default Javascript `Object`, which means functions like `hasOwnProperty`, `toString` will not be available. The extended version allows more data flexibility, but it is outmatched by JSON.

Reviewer 2: With the addition of this line, I do not think it is necessary to change the encoding to use the extended parser.

Author: Hi @ShaunSHamilton I have reverted the encoding to `false` as it was there originally

> When using `extended=false`, values can be only strings or arrays. The object returned when using `querystring` does not prototypically inherit from the default Javascript `Object`, which means functions like `hasOwnProperty`, `toString` will not be available. The extended version allows more data flexibility, but it is outmatched by JSON.

Reviewer 3: 😛 

```suggestion
When using `extended=false`, values can be only strings or arrays. The object returned when using `querystring` does not prototypically inherit from the default JavaScript `Object`, which means functions like `hasOwnProperty`, `toString` will not be available. The extended version allows more data flexibility, but it is outmatched by JSON.
```
